test/unittest/fem/test_calcshape.cpp

```diff
@@ -0,0 +1,200 @@
+         break;
+      case 3:
+         arr.SetSize(15);
+
+         pt.Set3(  ip.x,   ip.y,   ip.z );  arr[idx++] = pt;
+         pt.Set3( -ip.x,   ip.y,   ip.z );  arr[idx++] = pt;
+         pt.Set3(  ip.x,  -ip.y,   ip.z );  arr[idx++] = pt;
+         pt.Set3( -ip.x,  -ip.y,   ip.z );  arr[idx++] = pt;
+         pt.Set3(  ip.x,   ip.y,  -ip.z );  arr[idx++] = pt;
+         pt.Set3( -ip.x,   ip.y,  -ip.z );  arr[idx++] = pt;
+         pt.Set3(  ip.x,  -ip.y,  -ip.z );  arr[idx++] = pt;
+         pt.Set3( -ip.x,  -ip.y,  -ip.z );  arr[idx++] = pt;
+         pt.Set3(1+ip.x,   ip.y,   ip.z );  arr[idx++] = pt;
+         pt.Set3(  ip.x, 1+ip.y,   ip.z );  arr[idx++] = pt;
+         pt.Set3(1+ip.x, 1+ip.y,   ip.z );  arr[idx++] = pt;
+         pt.Set3(  ip.x,   ip.y, 1+ip.z );  arr[idx++] = pt;
+         pt.Set3(1+ip.x,   ip.y, 1+ip.z );  arr[idx++] = pt;
+         pt.Set3(  ip.x, 1+ip.y, 1+ip.z );  arr[idx++] = pt;
+         pt.Set3(1+ip.x, 1+ip.y, 1+ip.z );  arr[idx++] = pt;
+         break;
+   }
+}
+
+/**
+ * Tests fe->CalcShape() over a grid of IntegrationPoints
+ * of resolution res. Also tests at integration poins
+ * that are outside the element.
+ */
+void TestCalcShape(FiniteElement* fe, int res)
+{
+   IntegrationPoint ip;
+   int dim = fe->GetDim();
+
+   Vector weights( fe->GetDof() );
+
+   // Get a uniform grid or integration points
+   RefinedGeometry* ref = GlobGeometryRefiner.Refine( fe->GetGeomType(), res);
+   const IntegrationRule& intRule = ref->RefPts;
+
+   int npoints = intRule.GetNPoints();
+   for (int i=0; i < npoints; ++i)
+   {
+      // Get the current integration point from intRule
+      IntegrationPoint pt = intRule.IntPoint(i);
+
+      // Get several variants of this integration point
+      // some of which are inside the element and some are outside
+      Array<IntegrationPoint> ipArr;
+      GetRelatedIntegrationPoints( pt, dim, ipArr );
+
+      // For each such integration point check that the weights
+      // from CalcShape() sum to one
+      for (int j=0; j < ipArr.Size(); ++j)
+      {
+         IntegrationPoint& ip = ipArr[j];
+         fe->CalcShape(ip, weights);
+         REQUIRE( weights.Sum() == Approx(1.) );
+      }
+   }
+}
+
+
+TEST_CASE("CalcShape for several Lagrange FiniteElement instances",
+          "[Lagrange1DFiniteElement]"
+          "[BiLinear2DFiniteElement]"
+          "[BiQuad2DFiniteElement]"
+          "[LagrangeHexFiniteElement]")
+{
+   int maxOrder = 5;
+   int resolution = 10;
+
+   SECTION("Lagrange1DFiniteElement")
+   {
+      for (int order =1; order <= maxOrder; ++order)
+      {
+         std::cout << "Testing Lagrange1DFiniteElement::CalcShape() "
+                   << "for order " << order << std::endl;
+         Lagrange1DFiniteElement fe(order);
+         TestCalcShape(&fe, resolution);
+      }
+   }
+
+   SECTION("BiLinear2DFiniteElement")
+   {
+      std::cout << "Testing BiLinear2DFiniteElement::CalcShape()" << std::endl;
```

Comment:
Do we want outputs like these, or do we want the tests to be silent and rely on outputs from runs like `cd test/unittest; ./test -d yes`

Reply:
It might make sense to convert these to logging macros, which appears to scope the output, and (possibly) only output when there are errors.
See: https://github.com/catchorg/Catch2/blob/master/docs/logging.md